pkg/gojail/createchildjail.go

```diff
@@ -0,0 +1,142 @@
+package gojail
+
+import (
+	"errors"
+	"os"
+	"syscall"
+	"unsafe"
+
+	"golang.org/x/sys/unix"
+)
+
+//go:norace
+func forkAndCreateChildJail(parentID, iovecs, niovecs, errbufptr uintptr, pipe int) (pid int, err syscall.Errno) {
+	var (
+		r1      uintptr
+		jid     int32
+		err1    syscall.Errno
+		sendmsg = false
+	)
+
+	r1, _, err1 = syscall.RawSyscall(syscall.SYS_FORK, 0, 0, 0)
```

Comment:
Is it possible to use pdfork here and expose process descriptors over the interface?  This would make it feasible to apply Capsicum sandboxing to `runj` in the future (I believe Ben Laurie added Capsicum support to the go core library some time ago).  It's always possible to convert from a process descriptor to a PID, but operations on PIDs are not permitted in sandboxed mode, whereas operations on process descriptors are enable everywhere.

Reply:
This would be to apply capsicum on the jailed process correct?

In that case here would be the wrong place to do so, as this function exists solely to create a jail as a child from another jail, with the intention being that the parent jail functions as a namespace of sorts. 
This is useful for k8s where pods expect a network namespace to be shared between its containers and I use it in my Docker port as a network namespace. The biggest advantage there is that this allows running Linux containers with networking support (as they would lack the FreeBSD networking configuration tools).

Reply:
To add: the forked child dies immediately after creating the jail, or failing to do so.

The fork is needed because a child jail can only be created from within the parent, and (rightly so) we cannot enter and leave a jail at will.

Reply:
> This would be to apply capsicum on the jailed process correct?

It would apply capsicum to any child process, though it's up to the consumer of this API when they enter capability mode.  If the parent exits immediately after this, then there is no benefit from doing so.  If the parent continues executing then it may wish to drop privileges at that point.  If the parent wishes to maintain a handle for monitoring the child, then it may wish to do so.

Either way, the APIs using process descriptors are more general and do not have the problems with data races that the PID-based APIs have and so should be preferred where there are not legacy API compatibility concerns.  

Did you validate that this functionality works with the `security.bsd.see_jail_proc` sysctl set to 0?

Reply:
Ok, I understand the scope now :) I haven't tested that yet. It works fine from the root system.

How does one wait on a procdesc FD? The documentation of pdfork & wait do not seem to mention a way to do so.